diff --git a/CoreFoundation/Base.subproj/CFSystemDirectories.c b/CoreFoundation/Base.subproj/CFSystemDirectories.c
index a4809037..83087094 100644
--- a/CoreFoundation/Base.subproj/CFSystemDirectories.c
+++ b/CoreFoundation/Base.subproj/CFSystemDirectories.c
@@ -26,14 +26,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <pwd.h>
-#include <sysdir.h>
-
-// For now, CFSystemDirectories SPIs are exactly equivalent to (or at least a subset of) sysdir's. NSSearchPath* APIs in Foundation are not a subset of sysdir, so don't attempt to push that functionality down here without accommodating the differences.
-#define CFSearchPathToSysdir(dir) ((sysdir_search_path_directory_t)dir)
-#define CFSearchPathDomainMaskToSysdir(mask) ((sysdir_search_path_domain_mask_t)domainMask)
+#include <NSSystemDirectories.h>

 CFSearchPathEnumerationState __CFStartSearchPathEnumeration(CFSearchPathDirectory dir, CFSearchPathDomainMask domainMask) {
-    return sysdir_start_search_path_enumeration(CFSearchPathToSysdir(dir), CFSearchPathDomainMaskToSysdir(domainMask));
+    return NSStartSearchPathEnumeration(dir, domainMask);
 }

 CFSearchPathEnumerationState __CFGetNextSearchPathEnumeration(CFSearchPathEnumerationState state, uint8_t *path, CFIndex pathSize) {
@@ -41,10 +37,10 @@ CFSearchPathEnumerationState __CFGetNextSearchPathEnumeration(CFSearchPathEnumer
     // NSGetNextSearchPathEnumeration requires a MAX_PATH size
     if (pathSize < PATH_MAX) {
         uint8_t tempPath[PATH_MAX];
-        result = sysdir_get_next_search_path_enumeration(state, (char *)tempPath);
+        result = NSGetNextSearchPathEnumeration(state, (char *)tempPath);
         strlcpy((char *)path, (char *)tempPath, pathSize);
     } else {
-        result = sysdir_get_next_search_path_enumeration(state, (char *)path);
+        result = NSGetNextSearchPathEnumeration(state, (char *)path);
     }
     return result;
 }
