diff --git a/CoreFoundation/Base.subproj/CoreFoundation.h b/CoreFoundation/Base.subproj/CoreFoundation.h
index b95bf265..86b491ff 100644
--- a/CoreFoundation/Base.subproj/CoreFoundation.h
+++ b/CoreFoundation/Base.subproj/CoreFoundation.h
@@ -91,10 +91,5 @@
 
 #include <CoreFoundation/CFUserNotification.h>
 
-#if !DEPLOYMENT_RUNTIME_SWIFT
-#include <CoreFoundation/CFXMLNode.h>
-#include <CoreFoundation/CFXMLParser.h>
-#endif
-
 #endif /* ! __COREFOUNDATION_COREFOUNDATION__ */
 
diff --git a/CoreFoundation/Base.subproj/ForFoundationOnly.h b/CoreFoundation/Base.subproj/ForFoundationOnly.h
index 1eb77df7..28db52af 100644
--- a/CoreFoundation/Base.subproj/ForFoundationOnly.h
+++ b/CoreFoundation/Base.subproj/ForFoundationOnly.h
@@ -870,5 +870,11 @@ CF_INLINE void _scoped_id_array_cleanup(id _Nonnull * _Nullable * _Nonnull mallo
     #endif
 #endif
 
+typedef pthread_t _CFThreadRef;
+typedef pthread_attr_t _CFThreadAttributes;
+typedef pthread_key_t _CFThreadSpecificKey;
+
+CF_CROSS_PLATFORM_EXPORT int _CFThreadSetName(_CFThreadRef _Nonnull thread, const char *_Nonnull name);
+
 #endif /* ! __COREFOUNDATION_FORFOUNDATIONONLY__ */

diff --git a/CoreFoundation/Parsing.subproj/CFPropertyList_Internal.h b/CoreFoundation/Parsing.subproj/CFPropertyList_Internal.h
index 76f247c6..fbd91a92 100644
--- a/CoreFoundation/Parsing.subproj/CFPropertyList_Internal.h
+++ b/CoreFoundation/Parsing.subproj/CFPropertyList_Internal.h
@@ -10,7 +10,7 @@
 #if !defined(__COREFOUNDATION_CFPROPERTYLIST_INTERNAL__)
 #define __COREFOUNDATION_CFPROPERTYLIST_INTERNAL 1
 
-#include "CFBase.h"
+#include <CoreFoundation/CFBase.h>
 
 /// Limit for the max recursion depth to avoid unbounded stack explosion when
 /// parsing a crafted plist during validation of an object graph and during reading.
